### PR TITLE
Instructions on how to run the Istio e2e

### PR DESCRIPTION
**Pull request by john-a-joyce**
This README describes the CNI tests and
the specific procedure to run the e2e
tests.

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+In order to run the e2e tests in a local environment first confirm:
+
+1. That you can run any of the Istio e2e tests in your local environment
+
+2. That your local environment can support the CNI plugin
+
+To run the Istio e2e test first clone the Istio repo in your local environment.  Then there are two options to run the tests.
+
+1. Run the Istio make target  ```sh make e2e_simple_cni```
```
Comment (venilnoronha) [inline test/README.md]: Need a line break before the code block. Something like:

<pre>
```console
$ make e2e_simple_cni
```
</pre>

File: test/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+2. Run any of the Istio e2e targets after setting up a few environmental variables:
```
Comment (venilnoronha) [inline test/README.md]: nit: environmental -> environment

File: test/README.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		1. istio-cni.tag
+		2. istio-cni.hub
+		3. istio-cni.excludeNamespaces
+
+The value for EXTRA_HELM_SETTINGS will depend on your specific environment.
+
+The tag should be set to the value you have used when pushing you CNI image.
+The hub should be set to the location you have pushed your CNI image.
+Istio in most cases runs the e2e tests in the istio-system namespace and therefore the excludedNamespaces must be set to NULL.
+The e2e tests normally use istioctl to inject the sidecar and it is necessary to use a conffigmap without the initContianers section.
```
Comment (venilnoronha) [inline test/README.md]: nit: conffigmap -> ConfigMap

File: test/README.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+If the tag and hub is not set the testwill use the latest default hub and tag values checking into the istio/cni rep.
```
Comment (venilnoronha) [inline test/README.md]: nit: is not set -> is not set,
nit: testwill -> test will
nit: rep. -> repo.

File: test/README.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+	2. E2E_ARGS=--kube_inject_configmap=istio-sidecar-injector
+	3. EXTRA_HELM_SETTINGS
+		1. istio-cni.tag
+		2. istio-cni.hub
+		3. istio-cni.excludeNamespaces
+
+The value for EXTRA_HELM_SETTINGS will depend on your specific environment.
+
+The tag should be set to the value you have used when pushing you CNI image.
+The hub should be set to the location you have pushed your CNI image.
```
Comment (venilnoronha) [inline test/README.md]: It'd be good to add some example values for these parameters.

File: test/README.md (hunk 3 above)
Comment (venilnoronha) [inline test/README.md]: A general comment: it'd be good to highlight keywords like `istio-system` and `istioctl` via ``.

File: test/README.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+```console
+export ENABLE_ISTIO_CNI=true
+export E2E_ARGS=--kube_inject_configmap=istio-sidecar-injector
+export EXTRA_HELM_SETTINGS=--set istio-cni.excludeNamespaces={}  --set istio-cni.tag=$YOUR_CNI_TAG) --set istio-cni.hub=$YOUR_CNI_HUB
+```
+The value for EXTRA_HELM_SETTINGS will depend on your specific environment.
+
+The tag ($YOUR_CNI_TAG) should be set to the $TAG value you used when you build your CNI image.
+The hub ($YOUR_CNI_HUB) should be set to the location you used when you build your CNI image.
+Istio in most cases runs the e2e tests in the ```istio-system``` namespace and therefore the excludedNamespaces must be set to NULL.
```
Comment (venilnoronha) [inline test/README.md]: nit: excludedNamespaces -> \`excludedNamespaces\`
nit: NULL -> \`NULL\`

File: test/README.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+The e2e tests normally use ```istioctl``` to inject the sidecar and it is necessary to use a ConfigMap without the ```initContainers``` section.
```
Comment (venilnoronha) [inline test/README.md]: nit: ConfigMap -> \`ConfigMap\`

**Comment by john-a-joyce** [comment]
Dummy comment

File: test/README.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+Depending on your environment you may need to override other default settings.  Any additional override settings can be added via the ```EXTRA_HELM_SETTINGS```
+
+If the tag and hub is not set, the test will use the latest hub and tag values checked into the istio/cni repository.  The default hub and tag values are fine to use if you do not want to build your own CNI images.
```
Comment (venilnoronha) [inline test/README.md]: nit: tag -> \`tag\`
nit: hub -> \`hub\`

**Pull request merged**